File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/cached/persistence/IDiskFileCachePersistence.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.cached.persistence;

import com.lasthopesoftware.bluewater.client.library.items.media.files.cached.repository.CachedFile;
import com.namehillsoftware.handoff.promises.Promise;

import java.io.File;

public interface IDiskFileCachePersistence {
	Promise<CachedFile> putIntoDatabase(final String uniqueKey, final File file);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/files/cached/persistence/IDiskFileCachePersistence.kt
package com.lasthopesoftware.bluewater.client.browsing.files.cached.persistence

import com.lasthopesoftware.bluewater.client.browsing.files.cached.repository.CachedFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise
import java.io.File

interface IDiskFileCachePersistence {
    fun putIntoDatabase(libraryId: LibraryId, uniqueKey: String, file: File): Promise<CachedFile?>
}
